mm2src/lp_ordermatch.rs

```diff
@@ -2021,31 +1903,38 @@ impl Orderbook {
     fn process_keep_alive(
         &mut self,
         from_pubkey: &str,
-        pairs: Vec<String>,
         message: new_protocol::PubkeyKeepAlive,
     ) -> Option<OrdermatchRequest> {
         let pubkey_state = pubkey_state_mut(&mut self.pubkeys_state, from_pubkey);
-        for pair in pairs {
-            if !pubkey_state.order_pairs_trie_roots.contains_key(&pair)
-                && self
-                    .topics_subscribed_to
-                    .contains_key(&orderbook_topic_from_ordered_pair(&pair))
+
+        let mut trie_roots_to_request = HashMap::new();
+        for (alb_pair, trie_root) in message.trie_roots {
```

Comment:
Here we iterate over maker's trie_roots, but he could (in modified application) send not all of them (just remove orders or some pairs), and we will not find out about it.
I offer to remove not found pairs in maker KeepAlive

Reply:
Yes, it is fine to remove them here.